### PR TITLE
platforms/vita

### PR DESCRIPTION
**Pull request by kouta-kun**
Hi! I made the following changes to the code so it can run on PSVita. "m3_wasi_unstable_path_open" currently doesn't work because only daveeftw's musl library has an "openat" implementation but I couldn't compile VitaSDK with it. However I was able to 'borrow' his clock implementation (MIT licensed so it's all good i think). For now it just works like the hardcoded emscripten implementation, maybe later i'll add a wasm file chooser.

### CODE REVIEW

File: extra/fib32.wasm.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+unsigned char fib32_wasm[] = {
```
Comment (vshymanskyy) [inline extra/fib32.wasm.h]: `fib32.wasm.h` is already available in `source/extra`

File: platforms/vita/main.c [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    IM3Module module;
+    result = m3_ParseModule (env, &module, wasm, fsize);
+
+    result = m3_LoadModule (runtime, module);
+
+    /*
+    result = m3_LinkWASI (runtime->modules);
+    if (result) FATAL("m3_LinkWASI: %s", result);
+
```
Comment (vshymanskyy) [inline platforms/vita/main.c]: Your changes to WASI below are basically not used (WASI is not linked).

**Comment by vshymanskyy** [comment]
@kouta-kun will you have an ability to update your PR? Thanks!

**Comment by kouta-kun** [comment]
Well, there's a bit of a roadblock. It turns out the Vita kernel doesn't support replacing FDs, and since WASI programs directly access FDs I can't actually write the stdout to the screen so there's not much more I can do with it. The most that can be done out of it in VitaSDK is maybe using it for scripting in some game.

**Comment by vshymanskyy** [comment]
@kouta-kun thanks for explanation. WASI is not mandatory, it fact it's a very limited implementation of WASI here. Most embedded platforms don't rely on WASI, so I'd prefer disabling WASI for the Vita as well.

**Pull request closed without merging**